app/src/main/java/com.example.weatherapp/WeatherAdapter.kt

```diff
@@ -0,0 +1,64 @@
+package com.example.weatherapp
+
+import android.view.View
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.request.RequestOptions
+import com.example.weatherapp.data.model.ForecastModel
+import com.example.weatherapp.databinding.WeatherCardItemBinding
+import com.example.weatherapp.utils.getResource
+import com.example.weatherapp.utils.inflate
+import com.example.weatherapp.utils.loadUrl
+
+class WeatherAdapter(private val days: List<ForecastModel>) :
+    RecyclerView.Adapter<WeatherAdapter.WeatherViewHolder>() {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
+        WeatherViewHolder(parent.inflate(R.layout.weather_card_item, false))
+
+
+    override fun onBindViewHolder(holder: WeatherViewHolder, position: Int) {
+        holder.bind(days.elementAt(position))
+    }
+
+    override fun getItemCount(): Int = days.size
+
+    class WeatherViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        private val binding = WeatherCardItemBinding.bind(itemView)
+        fun bind(item: ForecastModel) {
+            binding.apply {
+                binding.weatherImageView.loadUrl(
+                    StringBuilder().append(URL)
+                        .append(item.currentWeather.first().weatherIconId)
+                        .append(FORMAT)
+                        .toString(),
+                    RequestOptions().placeholder(R.drawable.default_image)
+                )
+                weatherDescription.text =
+                    item.currentWeather.first().weatherConditionDescription.toUpperCase()
+                weatherForecastDate.text = item.dateAndTime
+                itemView.getResource(
+                    weatherMinimumTemperature, StringBuilder().append(MINIMUM_TEMPERATURE)
+                        .append(item.temperature.minimumTemperature)
+                )
+                itemView.getResource(
+                    weatherMaximunTemperature, StringBuilder().append(MAXIMUM_TEMPERATURE)
+                        .append(item.temperature.maximumTemperature)
+                )
+                itemView.getResource(
+                    weatherThermalSensation, StringBuilder().append(FEELS_LIKE)
+                        .append(item.temperature.thermalSensation)
+                )
+            }
+
+        }
+    }
+
+    companion object {
+        private const val MINIMUM_TEMPERATURE = "MIN: "
```

Comment:
put these values that you show on the screen into a string res

Reply:
Done